### DISCUSSION TITLE
Getting current language in a ts file

### DISCUSSION THREAD

**Discussion by jonacds** [i18next/i18next#2041] [Q&A]
I'm trying to use the current language of the application in some utility functions in a typescript file.

The problem is that the only way I can get the current language is if I directly access the instance ex: myObject[ii8n.language].

If I declare a constant with the value of ii8n.language it is not updated in the function. How can I avoid having to always use ii8n.language?

Thanks in advance

**Comment by duoluodexiaoxiaoyuan** [comment]
const { t, i18n } = useTranslation();

example
```javascript
import { useTranslation } from 'react-i18next';

function MyComponent() {
  const { i18n } = useTranslation();
  
  let style;
  if (i18n.language === 'en') {
    style = { width: '500px' }
  } else {
    style = { width: '300px' };
  }

  return <Table style={style} />; 
}
```

hope can help u

accross my use , i find  when pages  lang change, it's can trigger
